kv/txn_coord_sender.go

```diff
@@ -730,7 +730,8 @@ func (tc *TxnCoordSender) updateState(ctx context.Context, ba roachpb.BatchReque
 			panic("no replica set in header on uncertainty restart")
 		}
 		newTxn.Update(pErr.GetTxn())
-		newTxn.CertainNodes.Add(t.NodeID)
+		// No more restarts for this node for anything after ExistingTimestamp.
+		newTxn.UpdateUncertainty(t.NodeID, t.ExistingTimestamp)
```

Comment:
Are you sure this is what you want to do? This isn't the max timestamp for the node which was visited. It's just the timestamp in the uncertainty interval. Setting the max timestamp for `t.NodeID` to `t.ExistingTimestamp` will set it too low.

I think you should add code to this PR which _always_ returns the timestamp at the remote node when the request was received and then _always_ use that to set the max timestamp for that node in the distributed sender.